src/Features/Core/Portable/Snippets/SnippetProviders/AbstractSnippetProvider.cs

```diff
@@ -61,23 +62,19 @@ internal abstract class AbstractSnippetProvider : ISnippetProvider
         /// Determines if the location is valid for a snippet,
         /// if so, then it creates a SnippetData.
         /// </summary>
-        public async Task<SnippetData?> GetSnippetDataAsync(Document document, int position, CancellationToken cancellationToken)
+        public SnippetData? GetSnippetDataCore(SyntaxContext context, CancellationToken cancellationToken)
         {
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            if (syntaxFacts.IsInNonUserCode(syntaxTree, position, cancellationToken))
-            {
-                return null;
-            }
-
-            if (!await IsValidSnippetLocationAsync(document, position, cancellationToken).ConfigureAwait(false))
+            if (!IsValidSnippetLocation(context, cancellationToken))
             {
                 return null;
             }
 
             return new SnippetData(Description, Identifier, AdditionalFilterTexts);
         }
 
+        public Task<SnippetData?> GetSnippetDataAsync(Document document, int position, CancellationToken cancellationToken)
+            => throw new NotImplementedException();
```

Comment:
This will not be called for `AbstractSnippetProvider`s. Instead, `GetSnippetDataCore` is called.